engine/src/exec-pasteboard.cpp

```diff
@@ -639,28 +639,80 @@ void MCPasteboardSetClipboardOrDragData(MCExecContext& ctxt, MCNameRef p_index,
 	
 	MCTransferType t_type;
 	if (p_index == nil)
-        t_type = TRANSFER_TYPE_UNICODE_TEXT;
+        t_type = TRANSFER_TYPE_TEXT;
 	else
 		t_type = MCTransferData::StringToType(MCNameGetString(p_index));
 
 	if (t_type != TRANSFER_TYPE_NULL && p_data != nil)
 	{    
-        MCAutoValueRef t_data;
+        MCAutoValueRef t_value;
         bool t_success;
         t_success = true;
-        // MW-2014-03-12: [[ ClipboardStyledText ]] If styledText is being requested, then
-        //   convert the array to a styles pickle and store that.
-        if (t_type == TRANSFER_TYPE_STYLED_TEXT_ARRAY)
-        {
-            t_type =  TRANSFER_TYPE_STYLED_TEXT;
-            t_success = MCConvertStyledTextArrayToStyledText((MCArrayRef)p_data, (MCDataRef&)&(t_data));
-        }
-        else
-            t_data = p_data;
-        
+		
+		switch (t_type)
+		{
+			case TRANSFER_TYPE_TEXT:
+			case TRANSFER_TYPE_FILES:
+			{
+				// convert to MCStringRef
+				MCStringRef t_string;
+				t_success = ctxt.ConvertToString(p_data, t_string);
+				if (t_success)
+					t_value = t_string;
+				
```

Comment:
Assignment to an Auto class via = increments the reference count, so I think t_string here (and the other ones below) should be MCAutoStringRefs / MCAutoDataRefs.